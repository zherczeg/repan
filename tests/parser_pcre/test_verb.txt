# Test verbs

/(*F)(*FAIL:abc)(*:abc)(*ACCEPT:)/
    print

/_(*PRUNE)(*UNKNOWN_VERB:abc)_/
    print

/(*:)/
    print

/(*MARK)/
    print

/(*MARK=abc)/
    print

/(*COMMIT:abc/
    print

/(*SKIP=abc)/
    print

# Test PCRE initial options

/(*UCP)(*ANY)(*NO_DOTSTAR_ANCHOR)abc/u
    print

/(*F)(*UCP)/
    print

/(*NO_JIT)(*UNKNOWN_VERB:abc)/
    print

/(*LIMIT_DEPTH=1024)/
    print

/(*LIMIT_HEAP)/
    print

/(*LIMIT_MATCH=)/
    print

/(*UCP)/
    print

/(*UTF)/
    print

/\p{C}/
    print

/\N{STRING TERMINATOR}/
    print

# Test script run

/(*sr:abc)(*script_run:def)/
    print

/(*sr)/
    print

/(*script_run=abc)/
    print
