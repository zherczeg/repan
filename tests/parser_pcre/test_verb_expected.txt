# Test verbs

/(*F)(*FAIL:abc)(*:abc)(*ACCEPT:)/
    print
    /(*FAIL)(*FAIL:abc)(*MARK:abc)(*ACCEPT)/

/_(*PRUNE)(*UNKNOWN_VERB:abc)_/
    print
    Pattern parse failed: Unknown control verb
        /_(*PRUNE)[ HERE ]>>(*UNKNOWN_VERB:abc)_/

/(*:)/
    print
    Pattern parse failed: Argument for (*MARK) control verb is expected
        /(*:[ HERE ]>>)/

/(*MARK)/
    print
    Pattern parse failed: A colon (:) is expected
        /(*MARK[ HERE ]>>)/

/(*MARK=abc)/
    print
    Pattern parse failed: A colon (:) is expected
        /(*MARK[ HERE ]>>=abc)/

/(*COMMIT:abc/
    print
    Pattern parse failed: A right bracket is expected
        /(*COMMIT:abc[ HERE ]>>/

/(*SKIP=abc)/
    print
    Pattern parse failed: A right bracket is expected
        /(*SKIP[ HERE ]>>=abc)/

# Test PCRE initial options

/(*UCP)(*ANY)(*NO_DOTSTAR_ANCHOR)abc/u
    print
    /(*UCP)(*ANY)(*NO_DOTSTAR_ANCHOR)abc/

/(*F)(*UCP)/
    print
    Pattern parse failed: Unknown control verb
        /(*F)[ HERE ]>>(*UCP)/

/(*NO_JIT)(*UNKNOWN_VERB:abc)/
    print
    Pattern parse failed: Unknown control verb
        /(*NO_JIT)[ HERE ]>>(*UNKNOWN_VERB:abc)/

/(*LIMIT_DEPTH=1024)/
    print
    /(*LIMIT_DEPTH=1024)/

/(*LIMIT_HEAP)/
    print
    Pattern parse failed: An equals sign (=) is expected
        /(*LIMIT_HEAP[ HERE ]>>)/

/(*LIMIT_MATCH=)/
    print
    Pattern parse failed: A decimal unsigned integer is required
        /(*LIMIT_MATCH=[ HERE ]>>)/

/(*UCP)/
    print
    Pattern parse failed: UTF support needs to be enabled for this pattern
        /[ HERE ]>>(*UCP)/

/(*UTF)/
    print
    Pattern parse failed: UTF support needs to be enabled for this pattern
        /[ HERE ]>>(*UTF)/

/\p{C}/
    print
    Pattern parse failed: UTF support needs to be enabled for this pattern
        /[ HERE ]>>\p{C}/

/\N{STRING TERMINATOR}/
    print
    Pattern parse failed: UTF support needs to be enabled for this pattern
        /[ HERE ]>>\N{STRING TERMINATOR}/

# Test script run

/(*sr:abc)(*script_run:def)/
    print
    /((*script_run:abc)((*script_run:def)/

/(*sr)/
    print
    Pattern parse failed: A colon (:) is expected
        /(*sr[ HERE ]>>)/

/(*script_run=abc)/
    print
    Pattern parse failed: A colon (:) is expected
        /(*script_run[ HERE ]>>=abc)/
